app/models/notifications/mentioned.rb

```diff
@@ -13,19 +13,25 @@ def deleted_translation_key
     end
 
     def linked_object
-      target.post
+      target.mentions_container
     end
 
     def self.notify(mentionable, recipient_user_ids)
       actor = mentionable.author
 
-      mentionable.mentions.select {|mention| mention.person.local? }.each do |mention|
+      get_mentions(mentionable, recipient_user_ids).each do |mention|
         recipient = mention.person
-
-        next if recipient == actor || !(mentionable.public || recipient_user_ids.include?(recipient.owner_id))
-
         create_notification(recipient.owner_id, mention, actor).email_the_user(mention, actor)
       end
     end
+
+    def self.get_mentions(mentionable, recipient_user_ids)
```

Comment:
I think we shouldn't use `get_`, rubocop has a cop for this, but only for instance methods.